src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestSCMEvent.java

```diff
@@ -119,16 +119,15 @@ public String description() {
             int namespaceLength = originProjectPath.lastIndexOf("/");
             originProjectPath = originProjectPath.substring(0, namespaceLength);
             Map<Boolean, Set<ChangeRequestCheckoutStrategy>> strategies = request.getMRStrategies();
-            boolean fork = !getPayload().getUser().getUsername().equals(originOwner);
+            boolean fork = !getPayload().getObjectAttributes().getSourceProjectId().equals(getPayload().getObjectAttributes().getTargetProjectId());
             for (ChangeRequestCheckoutStrategy strategy : strategies.get(fork)) {
                MergeRequestSCMHead h = new MergeRequestSCMHead(
                                 "MR-" + m.getIid() + (strategies.size() > 1 ? "-" + strategy.name()
                                         .toLowerCase(Locale.ENGLISH) : ""),
                                 m.getIid(),
                                 new BranchSCMHead(m.getTargetBranch()),
                                 ChangeRequestCheckoutStrategy.MERGE,
-                                originProjectPath
-                                        .equalsIgnoreCase(source.getProjectPath())
+                                !fork
```

Comment:
Hurts my brain :laughing: 
Could be rewritten as
```
fork ? new SCMHeadOrigin.Fork(originProjectPath) : SCMHeadOrigin.DEFAULT,
```

Reply:
Fixed in 8fee844